addons/point_of_sale/static/src/js/Screens/ProductScreen/ProductScreen.js

```diff
@@ -140,6 +140,10 @@ odoo.define('point_of_sale.ProductScreen', function(require) {
                 }
             }
 
+            if (this.env.pos.db.product_packaging_by_barcode[base_code.code]) {
```

Comment:
We forgot to consider that `base_code` can be undefined
```suggestion
            if (base_code && this.env.pos.db.product_packaging_by_barcode[base_code.code]) {
```

Reply:
why not if `(this.env.pos.db.product_packaging_by_barcode[base_code.get('code', '')])`
It return `false` no?

Reply:
But there is no `get` method in js object.

Reply:
... yes ...